### PR TITLE
Proposed update/clarification to Writer.future

### PR DESCRIPTION
**Pull request by bradcray**
After retiring four futures related to name munging yesterday, I started looking at other naming-conflict related futures including this one.  The original description of the problem seemed unclear to me, so I poked around a bit, changed the name of the bug, and updated the description (preserving the original one for posterity).

### CODE REVIEW

**Comment by bradcray** [comment]
@hildeth : Would you review the changes/additions that I've proposed for this .future file?

File: test/modules/hilde/Writer.future [hunk 1]
```diff
@@ -1,4 +1,32 @@
-bug: Parameter type name resolves to punned module name.
+bug: User definition of 'Writer' module breaks internal modules
+
+When code like defines a module named 'Writer' like this (whether
```
Comment (hildeth) [inline test/modules/hilde/Writer.future]: "When code like defines" --> "When code like this defines"

**Comment by hildeth** [comment]
It's a bit surprising that munging causes this problem to go away.  But then I recall that the name decoration is applied only to user names.  The implementation is separately responsible for not conflicting with names known to the backend.
That is OK.  I would lobby for making the handling of internal modules "less special", but that relies on better namespace management than what we have now.  Baby steps.
Anyway, the changes are fine by me.

**Comment by bradcray** [comment]
My change didn't fix this test -- I was checking seemingly/superficially related issues to see why it didn't improve them and was just updating the description based on what I found.

Thanks.

**Pull request merged**